### DISCUSSION TITLE
Is it possible to revalidate a form after setting the defaults?

### DISCUSSION THREAD

**Discussion by swelham** [airjp73/rvf#99] [Q&A] [answered]
Hi. I am trying to implement a lookup that populates another form on the page. I have this functionally working using a loader function and `setFormDefaults`. However, if a validation message was displayed before making the lookup, it remains visible after the field has been populated.

Here's a quick example

https://user-images.githubusercontent.com/4869877/165094482-91d2cfc7-97b7-45db-aa9b-26a8334dfd2c.mov

I can't see any functions that can be called client-side to revalidate or at least clear all existing validations on the form. Is there something I'm missing here?

**Comment by airjp73** [comment, marked as answer]
Adding a form-wide `validate` or `clearErrors` is a feature that's on my radar. Hopefully I'll have time to add it soon. It might even be worth adding an `reinitialize` option similar to what formik has.

For now, for this use-case you could pass a `formRef` to the validated form and call `formRef.current.reset()` after the new default values have been populated. Or adding a `key` prop to the form that changes when the default values change would also work.

**Reply by swelham** [reply]
Thanks so much for the quick response and workaround.

Official support for this in the library would awesome!

I tried both of those suggestions and found the `formRef` option to work the best. I noticed when using the `key` approach, it caused an issue where subsequent re-validations didn't appear to run.

Thanks again 😃

**Reply by airjp73** [reply]
Just released v4.4.0 which adds `validate`, `clearAllErrors`, and `reset` to `useFormContext`.

**Reply by fergusmeiklejohn** [reply]
I don't think `reset()` made into v4.4.0

**Reply by airjp73** [reply]
Whoops, you're right. I'm publishing 4.4.1 now to fix that.
